lib/iris/_lazy_data.py

```diff
@@ -69,21 +69,40 @@ def as_lazy_data(data, chunks=_MAX_CHUNK_SIZE):
     if not is_lazy_data(data):
         if isinstance(data, ma.MaskedArray):
             data = array_masked_to_nans(data)
-            data = data.data
         data = da.from_array(data, chunks=chunks)
     return data
 
 
-def array_masked_to_nans(array, mask=None):
+def array_masked_to_nans(masked_array):
     """
-    Convert a masked array to an `ndarray` with NaNs at masked points.
+    Convert a masked array to a NumPy `ndarray` filled with NaN values. Input
+    NumPy arrays with no mask are returned unchanged.
     This is used for dask integration, as dask does not support masked arrays.
-    Note that any fill value will be lost.
+
+    Args:
+
+    * masked_array:
+        A NumPy array. If masked, the masked points will be filled with
+        `np.nan` values.
+
+    Returns:
+        The input array if unmasked, or a NaN-filled masked array of
```

Comment:
Ah yes, a clunky description of what is happening. 

"Masked points are set to `nan`" might be better? I'll also add the bit about the "normal" array too.